components/skeleton/test/skeleton-group.visual-diff.html

```diff
@@ -0,0 +1,228 @@
+<!DOCTYPE html>
+<html lang="en">
+	<head>
+		<link rel="stylesheet" href="../../../test/styles.css" type="text/css">
+		<script type="module">
+			import '../../typography/typography.js';
+			import '../../collapsible-panel/collapsible-panel.js';
+			import '../../collapsible-panel/collapsible-panel-group.js';
+			import '../../inputs/input-checkbox.js';
+			import '../../inputs/input-text.js';
+			import '../../list/list.js';
+			import '../../list/list-item.js';
+
+			import { html, LitElement } from 'lit';
+			import { SkeletonGroupMixin } from '../skeleton-group-mixin.js';
+
+			class SkeletonGroupTestWrapper extends SkeletonGroupMixin(LitElement) {
+				render() {
+					return html`<slot></slot>`;
+				}
+			}
+			customElements.define('d2l-skeleton-group-test-wrapper', SkeletonGroupTestWrapper);
+		</script>
+		<style>
+			.visual-diff {
+				width: 300px;
+			}
+			d2l-input-text,
+			d2l-list {
+				margin-bottom: 0.6rem;
+			}
+		</style>
+		<title>d2l-skeleton-group</title>
+		<meta name="viewport" content="width=device-width, initial-scale=1.0">
+		<meta charset="UTF-8">
+	</head>
+	<body class="d2l-typography">
+		<div class="visual-diff" id="all-skeleton">
+			<d2l-skeleton-group-test-wrapper>
+				<d2l-collapsible-panel-group>
+					<d2l-collapsible-panel skeleton panel-title="Title 1"></d2l-collapsible-panel>
+					<d2l-collapsible-panel skeleton panel-title="Title 2"></d2l-collapsible-panel>
+					<d2l-collapsible-panel skeleton panel-title="Title 3"></d2l-collapsible-panel>
+				</d2l-collapsible-panel-group>
```

Comment:
Thoughts about testing the "skeleton group" portion of collapsible panel, list, table, etc. within the existing visual-diff tests for those components? It's essentially just another state they can be in.

And then most of these tests could be removed and could test skeleton-group-specific things like the adding/removing stuff, and they wouldn't necessarily need to reference other core components at all to do so.

My main concern is that we'll change collapsible panel in a minor way and these tests will "fail" unnecessarily.

Reply:
Hmm yeah, I could see lots of these living with collapsible panel visual diffs. Not exactly sure where that line would be -- a lot of these tests are specific to what happens in specific "skeleton group" situations. (1 skeleton element, removing the last skeleton element, etc). Would those all go with collapsible panel?

Reply:
Hmm yeah I'm flip-flopping a bit on it to be honest. I don't think skeleton groups needs tests for _every_ component that we make a group, in the same way it doesn't have tests for all (or any) of the components that can be skeletons. 

But also list/collapsible-panel-group shouldn't really need tests for skeleton group either since it's not really something they can "mess up" -- they just extend the mixin and they're done.

So yeah, I guess maybe leave this as-is for now? I think ideally skeleton group would have a few "test components" for testing all the things it needs to do and wouldn't be relying on any other core components... that way if those core components change, the skeleton group tests don't fail. But I also don't think it's worth it spending time re-writing these in that way!